### PR TITLE
JIT: Support more thrust functions

### PR DESCRIPTION
**Pull request by asi1024**
Part of PFN summer internship project made by @TsutsuiMasayoshi (2/4)
This PR supports `thrust::{copy, find, mismatch, sort, sort_by_key}`.

### CODE REVIEW

File: cupyx/jit/thrust.py [hunk 1]
```diff
@@ -18,6 +18,24 @@ def call(self, env, *args, **kwargs):
 device = _internal_types.Data('thrust::device', _cuda_types.Unknown())
 
 
+@_wrap_thrust_func(['thrust/copy.h', 'thrust/execution_policy.h'])
+def copy(env, exec_policy, first, last, result):
+    """Sorts the elements in [first, last) into ascending order.
+    """
```
Comment (emcastillo) [inline cupyx/jit/thrust.py:24]: Is this docstring correct?

File: cupyx/jit/thrust.py [hunk 2]
```diff
@@ -32,3 +50,76 @@ def count(env, exec_policy, first, last, value):
     args = [exec_policy, first, last, value]
     params = ', '.join([_internal_types.Data.init(a, env).code for a in args])
     return _internal_types.Data(f'thrust::count({params})', _cuda_types.int32)
+
+
+@_wrap_thrust_func(['thrust/find.h', 'thrust/execution_policy.h'])
+def find(env, exec_policy, first, last, value):
+    """Count the number of elements in [first, last) that equals to ``value``.
+    """
```
Comment (emcastillo) [inline cupyx/jit/thrust.py:58]: Same

File: cupyx/jit/thrust.py [hunk 3]
```diff
@@ -32,3 +50,76 @@ def count(env, exec_policy, first, last, value):
...
+    # TODO(asi1024): Typecheck for Comparable.
+    args = [exec_policy, first, last]
+    params = ', '.join([_internal_types.Data.init(a, env).code for a in args])
+    return _internal_types.Data(f'thrust::sort({params})', _cuda_types.void)
+
+
+@_wrap_thrust_func(['thrust/sort.h', 'thrust/execution_policy.h'])
+def sort_by_key(env, exec_policy, keys_first, keys_last, values_first):
+    """Sorts the elements in [first, last) into ascending order.
+    """
```
Comment (emcastillo) [inline cupyx/jit/thrust.py:111]: ditto

**Comment by leofang** [comment]
cc: @jrhemstad for vis

**Comment by emcastillo** [comment]
I will merge it, let's revert it if something bad happens

**Pull request merged**